actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -77,6 +77,10 @@ module RequestForgeryProtection
       config_accessor :log_warning_on_csrf_failure
       self.log_warning_on_csrf_failure = true
 
+      # Controls whether the Origin header is checked in addition to the CSRF token.
+      config_accessor :forgery_protection_origin_check
+      self.forgery_protection_origin_check = false
+
```

Comment:
We could leave it off by default for existing apps, but turn it on by default for new apps by adding an initializer to the generator.